src/hooks/useNotificationTypes.tsx

```diff
@@ -1026,6 +1030,202 @@ export const notificationTypes: NotificationTypeConfig[] = [
+        if (dydxAddress && depositAddresses && isDepositDialogOpen) {
+          setEnabled(true);
+          return;
+        }
+        setEnabled(false);
+      }, [dydxAddress, depositAddresses, activeDialog]);
+
+      // Stage 1: Watch for deposits from backend and update refs
+      useEffect(() => {
+        if (!dydxAddress || !depositStatus?.deposits.results || !depositAddresses) return;
+
+        const incomingDeposits = depositStatus.deposits.results;
+        const now = Date.now();
+        const sessionDuration = now - sessionStartTimestampRef.current;
+
+        const recentDeposits = incomingDeposits.filter((deposit) => {
+          const depositAge = now - new Date(deposit.created_at).getTime();
+
+          return depositAge >= 0 && depositAge <= sessionDuration;
+        });
+
+        if (recentDeposits.length === 0) return;
+
+        const prevDepositIds = prevDepositIdsRef.current;
+
+        const updatedDeposits = recentDeposits.filter((deposit) => !prevDepositIds.has(deposit.id));
+
+        if (updatedDeposits.length === 0) {
+          setNewDeposits([]);
+          return;
+        }
+
+        setNewDeposits(
+          updatedDeposits.toSorted(
+            (a, b) => new Date(a.created_at).getTime() - new Date(b.created_at).getTime()
+          )
+        );
+      }, [dydxAddress, depositStatus, depositAddresses, enabled]);
+
+      useEffect(() => {
+        if (newDeposits.length === 0) {
+          setEnabled(false);
+          return;
+        }
+
+        const now = Date.now();
+        newDeposits.forEach((deposit) => {
+          if (
+            now - lastNotificationTimeRef.current > NOTIFICATION_DEBOUNCE &&
+            !prevDepositIdsRef.current.has(deposit.id)
+          ) {
+            trigger({
+              id: `deposit-${deposit.id}`,
+              displayData: {
+                toastDuration: DEFAULT_TOAST_AUTO_CLOSE_MS * 2,
+                groupKey: 'DepositAddressEvents',
+                slotTitleLeft: (
+                  <Icon iconName={IconName.Signal} size="1.25rem" tw="text-color-success" />
+                ),
+                title: stringGetter({ key: STRING_KEYS.DEPOSIT_DETECTED_TITLE }),
+                body: stringGetter({
+                  key: STRING_KEYS.DEPOSIT_DETECTED_BODY,
+                  params: {
+                    AMOUNT: Number(deposit.amount).toFixed(2),
+                    ASSET: 'USDC',
+                  },
+                }),
+                toastSensitivity: 'foreground',
+                actionDescription: stringGetter({ key: STRING_KEYS.VIEW_TRANSACTION }),
+                renderActionSlot: () => {
+                  const chainInfo = CHAIN_INFO[deposit.chain_id];
+                  return (
+                    chainInfo?.explorerBaseUrl && (
+                      <Link
+                        href={`${chainInfo.explorerBaseUrl}/tx/${deposit.transaction_hash}`}
```

Comment:
this happens to work but is not a super safe way to determine the tx url. I think it is ok for now